x-pack/solutions/security/plugins/security_solution/public/data_view_manager/redux/listeners/data_view_selected.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataView, DataViewsServicePublic } from '@kbn/data-views-plugin/public';
+import type { AnyAction, Dispatch, ListenerEffectAPI } from '@reduxjs/toolkit';
+import isEmpty from 'lodash/isEmpty';
+import type { RootState } from '../reducer';
+import { scopes } from '../reducer';
+import { selectDataViewAsync } from '../actions';
+import { sharedDataViewManagerSlice } from '../slices';
+
+export const createDataViewSelectedListener = (dependencies: {
+  dataViews: DataViewsServicePublic;
+}) => {
+  return {
+    actionCreator: selectDataViewAsync,
+    effect: async (
+      action: ReturnType<typeof selectDataViewAsync>,
+      listenerApi: ListenerEffectAPI<RootState, Dispatch<AnyAction>>
+    ) => {
+      let dataViewByIdError: unknown;
+      let adhocDataViewCreationError: unknown;
+      let dataViewById: DataView | null = null;
+      let adHocDataView: DataView | null = null;
+
+      const state = listenerApi.getState();
+
+      const findCachedDataView = (id: string | null | undefined) => {
+        if (!id) {
+          return null;
+        }
+
+        const cachedAdHocDataView =
```

Comment:
cache here means it is saved in redux, right? not browser cache

Reply:
yes